sky/utils/config_utils.py

```diff
@@ -199,6 +199,13 @@ def merge_k8s_configs(
                         else:
                             base_config[key].append(new_volume)
             else:
-                base_config[key].extend(value)
+                # For other list values, merge lists and maintain uniqueness.
+                # Preserve order by keeping original items and appending any
+                # new item not already present.
+                result = list(base_config[key])
+                for item in value:
+                    if item not in result:
```

Comment:
```suggestion
                    if isinstance(item, dict) or item not in result:
```
got here while talking to @SeungjinYang about #5104. A list item in a k8s config may be a `map`, basically `dict` in python, and as `listMapKeys` are not directly available from the current code structure, maybe we could force-add all dict elements and just maintain uniqueness for scalar values? Maybe we can create another issue to address the `listMapKey` problem, but this change may be good for now to handle the other issues.